pkg/operator/operator.go

```diff
@@ -201,3 +206,35 @@ func SanitizeSTS(sts *appsv1.StatefulSet) {
 		sts.Spec.VolumeClaimTemplates[i].Kind = ""
 	}
 }
+
+// WaitForCacheSync synchronizes the informer's cache and will log a warning
+// every minute if the operation hasn't completed yet.
+func WaitForCacheSync(ctx context.Context, logger log.Logger, inf cache.SharedIndexInformer) bool {
```

Comment:
Do you mind to elaborate why we need to do this in a retry-fashion?

Reply:
I've updated the comment